### DISCUSSION TITLE
[Markdown] Links ending in underscore bug

### DISCUSSION THREAD

**Discussion by billy-the-ape** [community/community#18584] [Other Feature Feedback, Questions, & Ideas]
Hi, I'm wondering why links whose url ends in an underscore when rendered by GFM are not including the underscore in the link and directing users who click to the wrong address. This could be a potential pathway for phishing attacks.

For example: https://twitter.com/albert_

I notice this on many sites and tools that use GFM, and I'm just wondering why this is, and if anything can be done about it?

**Comment by MathiasReker** [comment]
`[https://twitter.com/albert_](https://twitter.com/albert_ "https://twitter.com/albert_")`

[https://twitter.com/albert_](https://twitter.com/albert_ "https://twitter.com/albert_")

**Reply by billy-the-ape** [reply]
I acknowledge that it is possible to render these links by using a specific syntax -- the end users actually using the tool are expecting to just paste any valid link into their comment and have it actually render correctly (as it does with every link _not_ ending in `_`) and not have to concern themselves with a robust understanding of GFM syntax

**Reply by LeoOnTheEarth** [reply]
You can see [GFM spec: 6.9 Autolinks (extension)](https://github.github.com/gfm/#extended-autolink-path-validation)
> Trailing punctuation (specifically, `?`, `!`, `.`, `,`, `:`, `*`, `_`, and `~`) will not be considered part of the autolink,  though they may be included in the interior of the link:
